pkg/util/extract_forwarded.go

```diff
@@ -0,0 +1,53 @@
+package util
+
+import (
+	"context"
+
+	"google.golang.org/grpc/metadata"
+)
+
+// ipAddressesKey is key for the GRPC metadata where the IP addresses are stored
+const ipAddressesKey = "x-forwarded-for"
```

Comment:
You should define a custom `type` (for the context key) to avoid collisions if the same key is defined in other packages (see https://golang.org/pkg/context/ doc).

Reply:
This uses the GRPC metadata which uses strings as keys. So adding an extra type won't add much as it has to be converted back to a string when adding to the metadata.MD type.
I've made ipAddressKey a lot longer by adding the package name so it's less likely to clash.

Reply:
I think I was drunk when I left this comment. You're right, sorry.